internal/server/kong/ws/payload.go

```diff
@@ -136,6 +138,12 @@ func (p *Payload) configForVersion(version string) (cacheEntry, error) {
 		}
 		// cache it
 		err = p.configCache.store(version, entry)
+		configUpdateProcessingDuration := time.Since(configUpdateProcessingStartTime).Seconds()
+		metrics.Histogram("data_plane_version_compatibility_duration_seconds", configUpdateProcessingDuration,
+			metrics.Tag{Key: "dp_version", Value: version},
+			metrics.Tag{Key: "status", Value: "success"},
```

Comment:
Did we want to update the status here to reflect whether `p.vc.ProcessConfigTableUpdates()` errored?

Additionally, this is capturing the duration of the config loading + storing as well, so it's technically more than just version compatibility, though at least right now, the majority of this code is focusing on versions compatibility right now. Perhaps well want to move this metric emitting code below the error handling logic for `p.vc.ProcessConfigTableUpdates()` & additionally move the `configUpdateProcessingStartTime` assignment right before the call. This is bit of a nit, as it shouldn't skew the metric results too badly,  but it could since this part of the code is blocked by a mutex.

Reply:
Excluded caching from processing metric. We can look at instrumenting cache hit/miss/set/error behavior in the future.
[bb17fd1](https://github.com/Kong/koko/pull/485/commits/bb17fd17013470fbc21afa2117546948ca2e9d0d)

Reply:
Awesome, looks good with me! Thanks for moving the logic around, I was mostly concerned about the mutex throwing off the validity of this metric, so appreciate that change!